src/httpClient.ts

```diff
@@ -176,89 +183,69 @@ export class HttpApiClient extends HttpClient {
             throw new Error("url must be a string");
         }
 
-        if (typeof data !== "object" && typeof data !== "undefined") {
-            throw new Error("data must be an object or undefined");
-        }
-
         if (typeof files !== "object" && typeof files !== "undefined") {
             throw new Error("files must be an object or undefined");
         }
 
-        if (typeof options !== "object" && typeof options !== "undefined") {
+        if (typeof serviceClientRequestOptions !== "object" && typeof serviceClientRequestOptions !== "undefined") {
             throw new Error("options must be an object or undefined");
         }
 
         // Validate method/parameter combinations
-        switch (method) {
-            case "GET":
-                if (data || files) {
-                    throw new Error("data and files cannot be defined on a GET call");
-                }
-                break;
-            case "HEAD":
-                if (data || files) {
-                    throw new Error("data and files cannot be defined on a HEAD call");
-                }
-                break;
-            case "DELETE":
-                if (files) {
-                    throw new Error("files cannot be defined on a DELETE call");
-                }
-                break;
+        if (method === "GET" || method === "HEAD") {
+            if (data !== undefined) {
+                throw new Error(`data cannot be defined on a ${method} call`);
+            }
+            if (files !== undefined) {
+                throw new Error(`files cannot be defined on a ${method} call`);
+            }
+        }
+        if (method === "DELETE" && files !== undefined) {
+            throw new Error("files cannot be defined on a DELETE call");
         }
 
-        options = options || {};
+        let options: any = serviceClientRequestOptions || {};
```

Comment:
We need to redefine the type, or it will throw a TS2339 error because it thinks options is of typeIServiceClientRequestOptions